nym-network-monitor/src/main.rs

```diff
@@ -172,6 +172,18 @@ async fn nym_topology_from_env() -> anyhow::Result<NymTopology> {
     let mut topology = NymTopology::new_empty(rewarded_set);
     topology.add_skimmed_nodes(&nodes);
 
+    let node_ids = topology.node_details().keys().cloned().collect::<Vec<_>>();
```

Comment:
that's the only **NOT NIT** I've noticed: you DON'T WANT to be doing this. because `node_details` will also give you **GATEWAYS** that can't operate as mixnodes

Reply:
Nice one, will look into chaining a filter in here, hopefully that info is in there